plugin/pkg/scheduler/factory/factory.go

```diff
@@ -478,48 +478,48 @@ func getNodeConditionPredicate() cache.NodeConditionPredicate {
 // scheduled.
 func (factory *ConfigFactory) createUnassignedNonTerminatedPodLW() *cache.ListWatch {
 	selector := fields.ParseSelectorOrDie("spec.nodeName==" + "" + ",status.phase!=" + string(api.PodSucceeded) + ",status.phase!=" + string(api.PodFailed))
-	return cache.NewListWatchFromClient(factory.Client, "pods", api.NamespaceAll, selector)
+	return cache.NewListWatchFromClient(factory.Client.Core().GetRESTClient(), "pods", api.NamespaceAll, selector)
```

Comment:
@caesarxuchao I think the best option, if you don't want people to break the glass, is to add a NewListWatch function to each client.